### PR TITLE
Add option to warn victims of teamkills via AutoTKWarn plugin

### PR DESCRIPTION
**Pull request by SeanWalsh95**
alternate solution to PR #250

### CODE REVIEW

File: squad-server/plugins/auto-tk-warn.js [hunk 1]
```diff
@@ -11,10 +11,20 @@ export default class AutoTKWarn extends BasePlugin {
 
   static get optionsSpecification() {
     return {
-      message: {
+      attacker_message: {
```
Comment (Thomas-Smyth) [inline squad-server/plugins/auto-tk-warn.js]: This should be camel case, i.e. `attackerMessage`.

File: squad-server/plugins/auto-tk-warn.js [hunk 2]
```diff
@@ -11,10 +11,20 @@ export default class AutoTKWarn extends BasePlugin {
...
         required: false,
-        description: 'The message to warn players with.',
+        description: 'The message to warn attacking players with.',
         default: 'Please apologise for ALL TKs in ALL chat!'
+      },
+      warn_victim: {
```
Comment (Thomas-Smyth) [inline squad-server/plugins/auto-tk-warn.js]: As above.

File: squad-server/plugins/auto-tk-warn.js [hunk 3]
```diff
@@ -11,10 +11,20 @@ export default class AutoTKWarn extends BasePlugin {
...
+        required: false,
+        description: 'Should we nottify the the victim, telling them that they were teamkilled via a warning message',
+        default: false
+      },
+      victim_message: {
```
Comment (Thomas-Smyth) [inline squad-server/plugins/auto-tk-warn.js]: As above.

File: squad-server/plugins/auto-tk-warn.js [hunk 4]
```diff
@@ -34,8 +44,11 @@ export default class AutoTKWarn extends BasePlugin {
   }
 
   async onTeamkill(info) {
-    if (!info.attacker) return;
-
-    await this.server.rcon.warn(info.attacker.steamID, this.options.message);
+    if (info.attacker) {
+      this.server.rcon.warn(info.attacker.steamID, this.options.attacker_message);
+    }
+    if (this.options.warn_victim && info.victim) {
```
Comment (Thomas-Smyth) [inline squad-server/plugins/auto-tk-warn.js]: We can probably do away with the `warn_victim` and just do it based on whether the message is defined or not - which means they do not need to be required.

```js
if (info.attacker && this.options.attackerMessage) {
  await this.server.rcon.warn(info.attackers.steamId, this.options.attackerMessage);
}

if (info.victim && this.options.victimMessage) {
  await this.server.rcon.warn(info.attackers.steamId, this.options.victimMessage);
}
```

**Review by Thomas-Smyth — CHANGES_REQUESTED** [review]

File: squad-server/plugins/auto-tk-warn.js (hunk 1 above)
Comment (SeanWalsh95) [reply squad-server/plugins/auto-tk-warn.js]: oops I've been writing python too long

File: squad-server/plugins/auto-tk-warn.js (hunk 4 above)
Comment (SeanWalsh95) [reply squad-server/plugins/auto-tk-warn.js]: I was wondering is the await necessary? wouldn't one warning request just block the other until it gets a response?

Comment (Thomas-Smyth) [reply squad-server/plugins/auto-tk-warn.js]: Yeah, that's true, we can probably do without them.

File: squad-server/plugins/auto-tk-warn.js [hunk 5]
```diff
@@ -11,10 +11,15 @@ export default class AutoTKWarn extends BasePlugin {
...
-      message: {
+      attackerMessage: {
         required: false,
-        description: 'The message to warn players with.',
+        description: 'The message to warn attacking players with.',
         default: 'Please apologise for ALL TKs in ALL chat!'
+      },
+      victimMessage: {
+        required: false,
+        description: 'The message that will be sent to the victim',
```
Comment (Thomas-Smyth) [inline squad-server/plugins/auto-tk-warn.js]: Can we add a full stop please?

**Review by Thomas-Smyth — CHANGES_REQUESTED** [review]

File: squad-server/plugins/auto-tk-warn.js [hunk 6]
```diff
@@ -11,10 +11,15 @@ export default class AutoTKWarn extends BasePlugin {
...
+      attackerMessage: {
         required: false,
-        description: 'The message to warn players with.',
+        description: 'The message to warn attacking players with.',
         default: 'Please apologise for ALL TKs in ALL chat!'
+      },
+      victimMessage: {
+        required: false,
+        description: 'The message that will be sent to the victim.',
+        default: 'You were killed by your own team.'
```
Comment (Thomas-Smyth) [inline squad-server/plugins/auto-tk-warn.js]: One last thing, I think we should default this to `null` so it's not enabled by default.

**Review by Thomas-Smyth — CHANGES_REQUESTED** [review]

File: squad-server/plugins/auto-tk-warn.js (hunk 6 above)
Comment (SeanWalsh95) [reply squad-server/plugins/auto-tk-warn.js]: yeah that was the original reason for the separate flag I'll set it to null and comment out the suggested message

**Comment by IgnisAlienus** [comment]
I'm closing out my PR, #250 , for this since this one is here.

**Pull request merged**